### ISSUE TITLE
Feature: HTTP/Websocket server

### ISSUE THREAD

**Issue by mickmister** [FigBug/Gin#21]
Currently the project exports a WebSocket client for use. Can this code be easily augmented to support hosting a WebSocket server?

**Comment by FigBug** [comment]
No, the client and server are quite different. Luckily, there are a lot of C++ libraries for servers (where there are only a few for clients). If you just need a simple webserver that supports websockets and https I recommend https://github.com/civetweb/civetweb
